Enable Atomics object unconditionally.

rreno opened the pull request:
#### 09e0997b7d2ef90d36b76ed9165033e5cedd7684
<pre>
Enable Atomics object unconditionally.
<a href="https://bugs.webkit.org/show_bug.cgi?id=243342">https://bugs.webkit.org/show_bug.cgi?id=243342</a>
&lt;rdar://97786341&gt;

Reviewed by Geoffrey Garen and Yusuke Suzuki.

The Atomics object is only being created and made available to the runtime when
SharedArrayBuffer is enabled. In practice this means Atomics is only available when
crossOriginIsolation is true.

Other major browsers expose the Atomics object regardless of the state of crossOriginIsolation.
Using Atomics without shared memory isn&apos;t particularly useful but the goal of this is
to improve our interopability with other browsers.

* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::createAtomicsProperty):
(JSC::JSGlobalObject::init):

Canonical link: <a href="https://commits.webkit.org/253137@main">https://commits.webkit.org/253137@main</a>
</pre>

geoffreygaren reviewed (approved):
r=me

Constellation commented on Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
```diff
@@ -1067,10 +1067,8 @@ void JSGlobalObject::init(VM& vm)
-    if (Options::useSharedArrayBuffer())
-        atomicsObject = AtomicsObject::create(vm, this, AtomicsObject::createStructure(vm, this, m_objectPrototype.get()));
+
+    AtomicsObject* atomicsObject = AtomicsObject::create(vm, this, AtomicsObject::createStructure(vm, this, m_objectPrototype.get()));
```
Since it is now always defined, I recommend defining it lazily like `Math` in the table (See `Math` in the table and `createMathProperty` function).

rreno replied:
Done. Thanks for the suggestion!

Constellation reviewed (approved):
r=me too.

Constellation commented on Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
```diff
@@ -1153,7 +1156,6 @@ capitalName ## Constructor* lowerName ## Constructor = featureFlag ? capitalName
     if (Options::useSharedArrayBuffer()) {
         putDirectWithoutTransition(vm, vm.propertyNames->SharedArrayBuffer, m_sharedArrayBufferStructure.constructor(this), static_cast<unsigned>(PropertyAttribute::DontEnum));
-        putDirectWithoutTransition(vm, vm.propertyNames->Atomics, atomicsObject, static_cast<unsigned>(PropertyAttribute::DontEnum));
     }
```
Let's remove `{` and `}` to align to the WebKit's coding style :)

rreno replied:
Done. If you're happy with this PR, could you please add the merge queue tag? I'm not yet a committer.

Constellation reviewed (commented):
Nit

rreno committed:
Enable Atomics object unconditionally.
https://bugs.webkit.org/show_bug.cgi?id=243342
<rdar://97786341>

Reviewed by Geoffrey Garen and Yusuke Suzuki.

The Atomics object is only being created and made available to the runtime when
SharedArrayBuffer is enabled. In practice this means Atomics is only available when
crossOriginIsolation is true.

Other major browsers expose the Atomics object regardless of the state of crossOriginIsolation.
Using Atomics without shared memory isn't particularly useful but the goal of this is
to improve our interopability with other browsers.

* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::createAtomicsProperty):
(JSC::JSGlobalObject::init):

Canonical link: https://commits.webkit.org/253137@main
